libs/gi/db/src/Database/DataManagers/OptConfigDataManager.ts

```diff
@@ -193,21 +192,23 @@ export class OptConfigDataManager extends DataManager<
       builds = []
       buildDate = 0
     } else {
-      builds = builds
-        .map((build) => {
-          if (typeof build !== 'object') return undefined
-          const { weaponId, artifactIds: artifactIdsRaw } =
-            build as GeneratedBuild
-          if (!this.database.weapons.get(weaponId)) return undefined
-          const artifactIds = objKeyMap(allArtifactSlotKeys, (slotKey) =>
+      builds = mapFilterInplace(builds, (build) => {
```

Comment:
If I'm understanding your implementation, `builds` won't change if there are no changes from validation, which means that validation is not generating a new `builds` array(that are the same as before). Is that correct?

Reply:
Yep